src/insertItem.ts

```diff
@@ -0,0 +1,449 @@
+        let cursorPos = insertedText.indexOf(lookFor);
+        pos = textEditor.document.positionAt(index + cursorPos + lookFor.length);
+        let newSelection = new vscode.Selection(pos, pos);
+        textEditor.selection = newSelection;
+        await commands.executeCommand('editor.action.insertSnippet', { name: resource.label });
+        textEditor.revealRange(new vscode.Range(pos, pos), vscode.TextEditorRevealType.Default);
+        return true;
+    }
+
+    private async getFunction(): Promise<Function> {
+        const outputType = await this.ui.showQuickPick(getItemType(), { placeHolder: 'Type of function output?' });
+        let parameters = await this.getFunctionParameters();
+        let functionDef = {
+            parameters: parameters,
+            output: {
+                type: outputType.value,
+                value: insertCursorText
+            }
+        };
+        return functionDef;
+    }
+
+    private async getFunctionParameters(): Promise<FunctionParameter[]> {
+        let parameterName: string;
+        let parameters = [];
+        do {
+            parameterName = await this.ui.showInputBox({ prompt: "Name of parameter? Leave empty for no more parameters" });
+            if (parameterName !== '') {
+                const parameterType = await this.ui.showQuickPick(getItemType(), { placeHolder: `Type of parameter ${parameterName}?` });
+                parameters.push({
+                    name: parameterName,
+                    type: parameterType.value
+                });
+            }
+
+        } while (parameterName !== '');
+        return parameters;
+    }
+
+    // tslint:disable-next-line:no-any
+    private async insertInArray(templatePart: Json.ArrayValue, textEditor: vscode.TextEditor, data: any): Promise<void> {
+        let index = templatePart.span.endIndex;
+        let text = JSON.stringify(data, null, '\t');
+        let indentedText = this.indent(`\r\n${text}\r\n`, 2);
+        await this.insertText(textEditor, index, `${indentedText}\t`, true);
+    }
+
+    private async getFunctionNamespace(): Promise<FunctionNameSpace> {
+        let namespaceName = await this.ui.showInputBox({ prompt: "Name of namespace?" });
+        let namespace = {
+            namespace: namespaceName,
+            members: await this.getFunctionMembers()
+        };
+        return namespace;
+    }
+
+    // tslint:disable-next-line:no-any
+    private async getFunctionMembers(): Promise<any> {
+        let functionName = await this.ui.showInputBox({ prompt: "Name of function?" });
+        let functionDef = await this.getFunction();
+        // tslint:disable-next-line:no-any
+        let members: any = {};
+        // tslint:disable-next-line:no-unsafe-any
+        members[functionName] = functionDef;
+        return members;
+    }
+
+    private async insertText(textEditor: vscode.TextEditor, index: number, text: string, setCursor: boolean = false): Promise<number> {
```

Comment:
Just FYI, I might later turn the insertion code into snippets, but no need to change it now.